packages/fuels-core/src/types/wrappers/transaction.rs

```diff
@@ -312,6 +362,32 @@ macro_rules! impl_tx_wrapper {
                 self.tx.witnesses()
             }
 
+            fn is_using_predicates(&self) -> bool {
+                self.is_using_predicates
+            }
+
+            fn precompute(&mut self, chain_id: &ChainId) -> Result<()> {
+                Ok(self.tx.precompute(chain_id)?)
+            }
+
+            fn estimate_predicates(
+                &mut self,
+                consensus_parameters: &ConsensusParameters,
+                gas_costs: &GasCosts,
+            ) -> Result<()> {
+                let gas_price = *self.tx.gas_price();
+                let gas_limit = *self.tx.gas_limit();
+                *self.tx.gas_price_mut() = 0;
+                *self.tx.gas_limit_mut() = consensus_parameters.max_gas_per_tx;
+
+                self.tx
+                    .estimate_predicates(consensus_parameters, gas_costs)?;
+                *self.tx.gas_price_mut() = gas_price;
+                *self.tx.gas_limit_mut() = gas_limit;
```

Comment:
```suggestion
                self.tx
                    .estimate_predicates(consensus_parameters, gas_costs)?;
```

Should have the same effect=)

Reply:
@xgreenx is this ☝️ expected behaviour?

Reply:
Yes, because you should have some coins at least to cover `min_fee` that already is known before predicate estimation. Estimating predicates also will increase the `min_fee` later=)